test/simple_select.test.js

```diff
@@ -0,0 +1,346 @@
+      map.fire('mouseup', makeMouseEvent(15, 15));
+
+      afterNextRender(() => {
+        var args = getFireArgs().filter(arg => arg[0] === 'draw.deselect');
+        t.equal(args.length, 1, 'should have one and only one selected end event');
+        if (args.length > 0) {
+          t.equal(args[0][1].featureIds.length, 1, 'should unselect only one feautre');
+          t.equal(args[0][1].featureIds[0], id, 'should unselect the feature we expect it to select');
+        }
+        cleanUp(() => t.end());
+      });
+    });
+  });
+
+  t.test('simple_select - click on an unselected feature', t => {
+    var id = Draw.add(cloneFeature('polygon'));
+    console.log(JSON.stringify(cloneFeature('polygon')));
+    Draw.changeMode('simple_select');
+
+    afterNextRender(() => {
+      map.fire.reset();
+      map.fire('mousedown', makeMouseEvent(50, 30));
+      map.fire('mouseup', makeMouseEvent(50, 30));
+
+      afterNextRender(() => {
+        var args = getFireArgs();
+        args = args.filter(arg => arg[0] === 'draw.select');
+        t.equal(args.length, 1, 'should have one and only one selected start event');
+        if (args.length > 0) {
+          t.equal(args[0][1].featureIds.length, 1, 'should select only one feautre');
+          t.equal(args[0][1].featureIds[0], id, 'should select the feature we expect it to select');
+        }
+        cleanUp(() => t.end());
+      });
+    });
+  });
+
+  t.test('simple_select - click on an selected feature with shift down', t => {
+    var id = Draw.add(cloneFeature('polygon'));
+    Draw.changeMode('simple_select', [id]);
+
+    afterNextRender(() => {
+      map.fire.reset();
+      map.fire('mousedown', makeMouseEvent(50, 30, true));
+      map.fire('mouseup', makeMouseEvent(50, 30, true));
+
+      afterNextRender(() => {
+        var args = getFireArgs();
+        args = args.filter(arg => arg[0] === 'draw.deselect');
+        t.equal(args.length, 1, 'should have one and only one selected end event');
+        if (args.length > 0) {
+          t.equal(args[0][1].featureIds.length, 1, 'should select only one feautre');
+          t.equal(args[0][1].featureIds[0], id, 'should select the feature we expect it to select');
+        }
+        cleanUp(() => t.end());
+      });
+    });
+  });
+
+  t.test('simple_select - delete selected features', t => {
+    var id = Draw.add(cloneFeature('polygon'));
+    Draw.changeMode('simple_select', [id]);
+    map.fire.reset();
+    Draw.trash();
+    var args = getFireArgs();
+    t.equal(args.length, 1, 'should have one and only one event');
+    t.equal(args[0][0], 'draw.deleted');
+    t.equal(args[0][1].featureIds.length, 1, 'should delete only one feautre');
+    t.equal(args[0][1].featureIds[0].id, id, 'should delete the feature we expect it to delete');
+
+    var selectedFeatures = Draw.getSelectedIds();
+    t.equal(selectedFeatures.length, 0, 'nothing should be selected anymore');
+    cleanUp(() => t.end());
+  });
```

Comment:
Pretty sure they are the same thing. We should have tests that confirm that delete emits a trash event maybe?